src/main/java/seedu/address/model/appointment/Date.java

```diff
@@ -0,0 +1,107 @@
+package seedu.address.model.appointment;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+import static seedu.address.model.util.DateTimeParser.INPUT_DATE_FORMATTER;
+
+import java.time.LocalDate;
+
+
+/**
+ * Represents the date of an appointment.
+ */
+public class Date {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Date should follow yyyy-MM-dd";
+    public static final String VALIDATION_REGEX =
+            "^(20[0-9]{2})-(0[1-9]|1[0-2])-(0[1-9]|[12][0-9]|3[01])$";
+    public final String value;
+
+    /** To facilitate filtering Appointments by date. **/
+    public final LocalDate localDate;
+
+    /**
+     * Constructs a {@code Date}.a
+     *
+     * @param date A valid date.
+     */
+    public Date(String date) {
+        requireNonNull(date);
+        checkArgument(isValidDate(date), MESSAGE_CONSTRAINTS);
```

Comment:
I think MESSAGE_CONSTRAINTS should be tweaked slightly based on isValidDate() implementation